### ISSUE TITLE
`null` value isn't retained on a List field marked @nullable

### ISSUE THREAD

**Issue by starction** [google/built_value.dart#1094]
my code:
```dart
abstract class MyClass implements Built<MyClass, MyClassBuilder> {
  @nullable
  @BuiltValueField(wireName: r'my_field')
  BuiltList<String> get myField;
}
```
is being generated to 

```
class MyClassBuilder implements Builder<MyClass, MyClassBuilder> {
  ListBuilder<String> _myField;
  ListBuilder<String> get myField =>
      _$this._myField ??= new ListBuilder<String>();
  set myField(ListBuilder<String> myField) =>
      _$this._myField = myField;
}
```

When I set value of `myField` to `null`, I was expecting it to return `null` on get, but it returns empty list instead (because of how the getter above is generated)

Appreciate any help on this matter! Thanks

**Comment by davidmorgan** [comment]
Yes, as you observed: getters auto-create builders so they never return `null`.

You can turn this behaviour off per class with the `@BuiltValue` annotation

https://pub.dev/documentation/built_value/latest/built_value/BuiltValue-class.html

or per field with the `@BuiltValueField` annotation

https://pub.dev/documentation/built_value/latest/built_value/BuiltValueField-class.html

in both cases set `autoCreateNestedBuilder: false` to change it.
